pegasus/sites.v3/code.org/public/private/k5survey/index.haml

```diff
@@ -22,119 +19,389 @@
+      venue_clean_total: venue_clean_total,
+      food_total: food_total,
+      tech_access_total: tech_access_total,
+      teacher_prepared_total: teacher_prepared_total,
+      useful_experience_total: useful_experience_total,
+      kept_attention_total: kept_attention_total,
+      useful_time_total: useful_time_total,
+      best_pd_total: best_pd_total,
+      logical_sequence_total: logical_sequence_total,
+      suitable_level_total: suitable_level_total,
+      equitable_total: equitable_total,
+      changed_beliefs_total: changed_beliefs_total,
+      current_knowledge_total: current_knowledge_total,
+      careers_knowledge_total: careers_knowledge_total,
+      comments: comments
+    }
+  end
 
-        # get data on students of the teacher
-        teacher_user_id = teacher[:id]
-        next unless teacher_user_id
+  if facilitator_id
+    workshops = DB[:forms].where(kind:'ProfessionalDevelopmentWorkshop', user_id:facilitator_id)
 
-        students = DASHBOARD_DB[:followers].
-          where(user_id: teacher_user_id).
-          join(:users, id: :student_user_id).
-          select(:users__id___id, :users__created_at___created_at)
+    query = "SELECT * FROM forms WHERE kind='K5ProfessionalDevelopmentSurvey' AND "
+    workshops.each do |workshop|
+      query += "data LIKE '%\"workshop_id_i\":\""
+      query += workshop[:id].to_s + "\"%\' OR "
+    end
+    query = query[0, query.length - 3]
 
-        if students.count > 0
-          lifetime = students.map{|s| (Time.now - s[:created_at]) / (60 * 60 * 24)}.reduce(:+) / students.count.to_f
+    individual_surveys = DB.fetch(query)
 
-          levels = students.map do |s|
-            DASHBOARD_DB[:user_levels].
-            where(user_id: s[:id]).
-            and("best_result >= #{ActivityConstants::MINIMUM_PASS_RESULT}").
-            count
-          end.reduce(:+) / students.count.to_f
-        else
-          lifetime = 0
-          levels = 0
-        end
+    individual_survey_totals = total_survey_data(individual_surveys)
+    all_surveys_totals = total_survey_data(all_surveys)
 
-        {
-          teacher_name: teacher[:name],
-          teacher_email: teacher[:email],
-          affiliate_name: affiliate[:name],
-          affiliate_email: affiliate[:user_id],
-          students_count: students.count,
-          students_average_lifetime_days: lifetime.round,
-          students_average_levels_completed: levels.round(2)
-        }
-      end.compact
-    end.compact.flatten
+    total_attendees = 0
+    workshops.each do |workshop|
+      data = JSON.parse(workshop[:data]).merge(JSON.parse(workshop[:processed_data]))
+      if data['total_attendee_count_i']
+        total_attendees += data['total_attendee_count_i']
+      end
+    end
   end
-%link(rel='stylesheet' type='text/css' href='/css/learn_tabs.css')
+
+:javascript
+  $(document).ready(function() {
+    $('select').bind("change", function () {
```

Comment:
the event object gives you a target so you could make this a bit more compact eg:
````
            $('select#sections').change(function(ev) {
              window.location.href = ev.target.value;
            });
````

Could give the select an id or class and only bind yours, to avoid confusing whoever adds another dropdown to this page later